### PR TITLE
Model Bob's behaviour if Alice doesn't send over the secret

### PR DESCRIPTION
**Pull request by luckysori**
I need help figuring out how to test this, because our implementation of Alice behaves in the best possible way, so we can't easily simulate a malicious/unpredictable actor. 

I attempted to use `cfg(test)` but that's not part of the public API of the crate so it's not available in our integration tests. 

I also thought about adding some internal tests, but It would be annoying to have to duplicate all the test harness code written for integration tests.

There's also a path dependency because I had to add a call to `bitcoin-harness-rs`, so this PR won't be mergeable until I open a PR for that.

### CODE REVIEW

**Comment by D4nte** [comment]
> I need help figuring out how to test this, because our implementation of Alice behaves in the best possible way, so we can't easily simulate a malicious/unpredictable actor.

Shouldn't you give the choice to the consumer to send the secret (or not) and exercise the pseudo-option? I think it'd be useful to give this flexibility which in turns will allow to test.

I am not really on top of PTLC so.. either merge or wait Monday for @bonomat's review?

**Comment by luckysori** [comment]
> Shouldn't you give the choice to the consumer to send the secret (or not) and exercise the pseudo-option?

I don't think so. It's one thing to test the behaviour of a malicious actor, which I managed to do by re-organising the modules, but I think providing a public API that enables someone to misbehave is going too far.

**Comment by luckysori** [comment]
Hey, @tcharding. When merging this branch with your latest changes I could have missed some things. You may want to have a look at https://github.com/comit-network/thor/pull/53/commits/e315eb0d99fcd67dffab47207768757258345b7e just in case.

I did revert the change which removed the `ptlc` module prefix, because I think it makes it clearer that those transactions belong to protocols involving PTLCs. Apart from that I tried to maintain your changes.

**Comment by tcharding** [comment]
> Hey, @tcharding. When merging this branch with your latest changes I could have missed some things. You may want to have a look at [e315eb0](https://github.com/comit-network/thor/commit/e315eb0d99fcd67dffab47207768757258345b7e) just in case.

I had a look through the diff (as it was displayed via that GitHub link) and it looks ok to me. Not sure where you thought the problem was though.

> I did revert the change which removed the `ptlc` module prefix, because I think it makes it clearer that those transactions belong to protocols involving PTLCs. Apart from that I tried to maintain your changes.

No worries. Also don't worry if some things got accidentally reverted, I'll keep it in mind when review down the track :)

**Review by tcharding — APPROVED** [review]

**Comment by tcharding** [comment]
Woops, I came to the keyboard half asleep and hit approve on this without adding the note: ACK the first patch, which AFAICT, is the bulk of this PR?

File: thor/src/lib.rs [hunk 1]
```diff
@@ -351,38 +366,123 @@ impl Channel {
...
 
-            // Perform a channel update to merge PTLC output into Alice's balance output
+                    if median_time >= ptlc_nearing_expiry_time {
+                        return Result::<(), anyhow::Error>::Ok(());
+                    }
 
-            let balance_output_self = SplitOutput::Balance {
-                amount: ours,
-                address: self.final_address_self.clone(),
+                    tokio::time::delay_for(Duration::from_secs(1)).await;
```
[the remark this answers is not in the document]
Comment (luckysori) [reply thor/src/lib.rs]: Good points. I think this and the average blocktime used above need to depend on the network.

> why not just poll every minute?

If I poll every minute in the tests I think we would have very slow tests.

[the remark this answers is not in the document]
Comment (tcharding) [reply thor/src/lib.rs]: > If I poll every minute in the tests I think we would have very slow tests.

True that.

**Comment by luckysori** [comment]
All this was merged with #65.
